src/ucp/core/ucp_ep.c

```diff
@@ -1176,7 +1176,8 @@ void ucp_ep_err_pending_purge(uct_pending_req_t *self, void *arg)
 
     /* TODO: check for context->config.ext.proto_enable when all protocols are
      *       implemented, such as flush, AM/RNDV, etc */
-    if (req->flags & UCP_REQUEST_FLAG_PROTO_SEND) {
+    if (req->flags & (UCP_REQUEST_FLAG_PROTO_SEND |
+                      UCP_REQUEST_FLAG_PROTO_INITIALIZED)) {
```

Comment:
Not all protocols have to use UCP_REQUEST_FLAG_PROTO_INITIALIZED flag (for example bcopy or single)
so abort function of each protocol should check this flag individually, or maybe some of them will use common helper function

Reply:
Actually I used this flag for RNDV GET proto on receiver side to avoid adding new flag.
Please note that condition based on `req->flags` is a temporary and should be replaced with `context->config.ext.proto_enable` in short term and completely removed in long term.

Reply:
even when `config.ext.proto_enable` is removed, it's still not OK to use UCP_REQUEST_FLAG_PROTO_INITIALIZED flag in the **common** code. it's only OK to use it for specific protocols (such as RNDV GET)

Reply:
then what do think about to set `UCP_REQUEST_FLAG_PROTO_SEND` for rndv_get internal request? (I see that this flag is set only from API functions) or need to add some new flag?

Reply:
> then what do think about to set `UCP_REQUEST_FLAG_PROTO_SEND` for rndv_get internal request? (I see that this flag is set only from API functions) or need to add some new flag?

Yes, this is the right way (not setting UCP_REQUEST_FLAG_PROTO_SEND for rndv_get internal request is a bug)